src/components/Common/UI/Button/Button.test.tsx

```diff
@@ -38,4 +39,36 @@ describe('Button', () => {
     const button = screen.getByRole('button', { name: 'Error Button' })
     expect(button).toHaveAttribute('data-error', 'true')
   })
+
+  describe('Accessibility', () => {
+    const testCases = [
+      { name: 'default', props: { children: 'Default Button' } },
+      {
+        name: 'primary',
+        props: { variant: 'primary' as const, children: 'Primary Button' },
+      },
+      {
+        name: 'secondary',
+        props: { variant: 'secondary' as const, children: 'Secondary Button' },
+      },
+      {
+        name: 'tertiary',
+        props: { variant: 'tertiary' as const, children: 'Tertiary Button' },
+      },
+      { name: 'disabled', props: { isDisabled: true, children: 'Disabled Button' } },
+      { name: 'loading', props: { isLoading: true, children: 'Loading Button' } },
+      {
+        name: 'icon',
+        props: { variant: 'icon' as const, 'aria-label': 'Icon button', children: '×' },
+      },
+    ]
+
+    it.each(testCases)(
+      'should not have any accessibility violations - $name',
+      async ({ props }) => {
+        const { container } = renderWithProviders(<Button {...props} />)
+        await expectNoAxeViolations(container)
+      },
+    )
+  })
```

Comment:
Same here, wondering if we might consider not testing on props that only modify the styling like variant since this isn't going to be able to verify pure style concerns like color contrast? The matrix of all prop permutations for this could start getting pretty expansive/hard to maintain